imports/importers.py

```diff
@@ -171,9 +173,28 @@ def _import_layer(self, layer):
                     feature_data[model_field] = processor(
                         feature[shapefile_field].value
                     )
-            feature_id = feature_data.pop("id")
+            feature_id = feature_data.pop("id", None)
             geometry = feature.geom.geos
             # The geometries in shapefile must be using the same SRID as defined in settings
             geometry.srid = settings.SRID
             feature_data["geometry"] = geometry
-            Feature.objects.update_or_create(id=feature_id, defaults=feature_data)
+
+            if feature_id:
+                try:
+                    feature = Feature.objects.get(id=feature_id)
+                    for key, value in feature_data.items():
+                        setattr(feature, key, value)
+                    feature.save()
+                    count += 1
+                except Feature.DoesNotExist:
+                    msg = _(
+                        "Update feature failed: feature with id {0} does not exist"
+                    ).format(feature_id)
+                    import_log.append([messages.WARNING, msg])
+            else:
+                Feature.objects.create(**feature_data)
+                count += 1
```

Comment:
Hmm... I don't like this block here. It looks really messy in my eyes.

Is there anything we could do here to make it read a bit nicer?
For instance, skip the try-except block and use `.filter()`, that would remove
one level of indentation at least.

I don't know, might be just me, but it looks a bit much with the if-try-for block here.

Reply:
IMO, it is important to notify the user about wrong feature ids in the shapefileso to make sure that all changes included in the shapefiles are applied to the database. Using `.filter()` does not help iin this case.

I think we could remove one level of indentation by moving up the else block and exit loop with continue, but not sure if it's any better